### PR TITLE
fix: add support Jira API Cloud version

### PR DESCRIPTION
**Pull request by afdesk**
# Description

This pull request addresses a particular issue encountered when using the Jira Cloud API's serverInfo endpoint. Currently, in certain cases, the serverInfo response contains a version that appears to be peculiar or unconventional. As a result, this pull request proposes an important fix to prevent the usage of API 9 for Jira Cloud.

**upd**: 
the next `if` statement looks a bit flaky:
https://github.com/aquasecurity/go-jira/blob/358abd58fd26585814d9887f41ba3fd53aa73956/metaissue.go#L134

so I refactored it.

# Checklist

* [ ] Unit or Integration tests added
  * [ ] Good Path
  * [ ] Error Path
* [ ] Commits follow conventions described here:
  * [ ] [Conventional Commits 1.0.0](https://conventionalcommits.org/en/v1.0.0-beta.4/#summary)
  * [ ] [The seven rules of a great Git commit message](https://chris.beams.io/posts/git-commit/#seven-rules)
* [ ] Commits are squashed such that
  * [ ] There is 1 commit per isolated change
* [ ] I've not made extraneous commits/changes that are unrelated to my change.

### CODE REVIEW

**Comment by afdesk** [comment]
@simar7 @roeelivaqua could take a look at this PR? thanks!

**Comment by simar7** [comment]
do you know why the tests are failing?

File: metaissue.go [hunk 1]
```diff
@@ -109,29 +112,38 @@ func (s *IssueService) GetCreateMetaWithOptionsWithContextForJira9(ctx context.C
...
+	// Jira Cloud API returns a bit strange version number now (1001)
+	// now Jira Cloud API doesn't support 9
+	if version.DeploymentType == "Cloud" {
+		return false
 	}
 
 	// versionNumbers contains an array with 3 numbers: major, minor, patch.
 	// we need only a major number
-	return version.VersionNumbers[0]
+	if version.VersionNumbers[0] == 9 {
```
Comment (roeelivaqua) [inline metaissue.go]: use >= for future jira 10.x versions?

Comment (afdesk) [reply metaissue.go]: yes, It makes sense. updated.
my concern was about changing of Jira API for future, but let's try this way.
thanks

**Review by roeelivaqua — APPROVED** [review]
LGTM

**Review by simar7 — APPROVED** [review]

**Pull request merged**